views/Main.react.js

```diff
@@ -627,6 +637,35 @@ class MainPage extends Component {
       return <span style={{ color: data.lastSyncedColor }}>{data.synced}</span>
     }
   }
+  updateUserPreferences = (configuration) => {
+    let uid = this.props.user.uid
+    const createNewPreference = preparePreferences(
+      configuration,
+      this.state.preferences
+    )
+    return window
+      .fetch(`${basePath}/api/v1/users/${uid}/preferences`, {
+        method: 'POST',
+        headers: {
+          'Content-Type': 'application/json',
+        },
+        credentials: 'same-origin',
+        body: JSON.stringify({
+          preferences: createNewPreference,
+        }),
+      })
+      .then((res) => {
+        if (res.status === 200) {
+          this.setState({
+            preferences: createNewPreference,
+          })
+        } else {
+          this.setState({
+            preferences: createNewPreference,
+          })
+        }
+      })
+  }
```

Comment:
it's weird it doesn't work if it lives outside in fetchUtils, this will probably need to be refactored, and because Mike pointed out that the feature goes on a different page I've had to reset this file.